machine.go

```diff
@@ -40,6 +40,7 @@ type Machine struct {
 	DiscoKey    string
 	IPAddresses MachineAddresses
 	Name        string
+	Nickname    string
```

Comment:
I am a bit conflicted about the naming here; `Nickname` doesnt just sound completely right, in a way, what is now called `Name` should become `HostName` and then `Nickname` should be `Name` or maybe `GivenName`

Reply:
I agree `Nickname` is not a good name. I didn't want to change the name of an existing field. Shall I change the `Name` to `Hostname` and keep the db column name as `name`?

Reply:
I am slightly concerned that the migration software actually understands this tho... I would like to have that sorted. I personally am not interested in having naming things we are not happy with accumulate. 

I suspect that using something like `RenameColumn` (https://gorm.io/docs/migration.html) would just end in an "infinite" loop, renaming it every startup

I think we can do:
```suggestion
	GivenName    string
```

and `HostName`, and add a comment above both with how they are set, and what they mean.